lessons/codap-transformers3/langs/en-us/index.adoc

```diff
@@ -0,0 +1,74 @@
+= More Transformers: Transform and Build
+@description{Students learn more about _Transformers_, including `Transform Attribute`, `Build Attribute`.}
+
+[@lesson-intro-table]
+|===
+@lesson-prereqs{codap-transformers-1}
+| Lesson Goals
+| Students will be able to...
+
+* filter the Animals Dataset by species, fixed status, and age
+* transform a column in the Animals Dataset
+* add a column to the Animals Dataset
+
+| Student-facing Lesson Goals
+|
+
+* Let’s learn how to transform one table into another.
+
+| Materials
+|[.materials-links]
+
+@material-links
+
+|===
+
+
+== Transforming Columns @duration{10 minutes}
+
+=== Overview
+Students learn how to _transform columns_ using the `Transform Attribute` Transformer.
+
+=== Launch
+Suppose we want to _transform_ our table, converting `pounds` to `kilograms` or `weeks` to `days`. The `Transform Attribute` Transformer makes that possible!
+
+=== Investigate
+[.lesson-instruction]
+Complete the worksheet @printable-exercise{pages/codap-transform.adoc} in the @starter-file{animals} in CODAP.
+
+The `Transform Attribute` Transformer walks through the table, applying whatever expression it was given to each row. Whatever the expression produces for that row becomes the value of the column; we name that column when we create the Transformer. In the first example, we gave the Transformer `Age < 5` as its expression, so the new table replaced the age with an indication of whether each animal is young (`true`) or not (`false`).
+
+=== Synthesize
+Debrief with students. Ask them if they can think of a situation where they would want to use this. Some ideas:
+
+- A dataset from Europe might list everything in metric (centimeters, kilograms, etc), so we use `Transform Attribute` to convert to imperial units (inches, pounds, etc).
+- A dataset about schools might include columns for how many students are in the school and how many of those students identify as multi-racial. But when comparing schools of different sizes, what we really want is a column showing what _percentage_ of students identify as multi-racial. We could use `Transform Attribute` to compute that for every row in the table.
+
+== Building Columns @duration{10 minutes}
+
+=== Overview
+Students learn how to _build columns_, using the `Build Attribute` Transformer.
+
+=== Launch
+So far, we've used Transformers to _filter_ and to _transform an attribute_. The final Transformer we are exploring today is called `Build Attribute`. Can you guess what this Transformer does and how it might be similar to _and_ different from `Transform Attribute`?
```

Comment:
I think this "can you guess" is too hard for kids. How _could_ they guess? Instead, maybe give an example of where you want the transformed column but you DON'T want to lose the original. Present the problem that is solved by Build Attribute

Reply:
also a good call!